contrib/libev/ev.c

```diff
@@ -2146,7 +2146,8 @@ typedef struct
   #include "ev_wrap.h"
 
   static struct ev_loop default_loop_struct;
-  EV_API_DECL struct ev_loop *ev_default_loop_ptr = 0; /* needs to be initialised to make it a definition despite extern */
```

Comment:
This cannot be done in this way, I'm afraid. Please revert it as it will break so linking badly. I found no good way to fix it without breaking both static and dynamic linking and there are no ways to do it reliably with LTO, as far as I remember.

Reply:
Ok, from what I found looking on the web gcc should not give a warning as they list this type of declaration as an example syntax. But also it says extern does not reserve memory for storing a value?

Reply:
Yes, but this case is a notable exception. It's all about linking after all. Removing of external will make 2 or more default ev_loops which is potentially extremely dangerous (especialy in the case of inotify usage). Removing of static is also not an option as it should be apparently invisible symbol. Hence, we have a contradiction: symbol must be unique and hidden at the same time. There are some gcc extensions that allows that but I'd better leave it as it is defined in the upstream.

Reply:
Ok, I copied from an example where first they create an int and set a value, then declare it extern.